lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1039,55 +1079,153 @@ def configure_cpu_and_memory(self, vm_obj, vm_creation=False):
+                cdrom_spec.device.connectable.connected = (self.params["cdrom"]["type"] != "none")
+
+        elif not self.device_helper.is_equal_cdrom(vm_obj=vm_obj, cdrom_device=cdrom_devices[0],
+                                                   cdrom_type=self.params["cdrom"]["type"], iso_path=iso_path):
+            self.device_helper.update_cdrom_config(vm_obj, self.params["cdrom"], cdrom_devices[0], iso_path=iso_path)
+            cdrom_spec = vim.vm.device.VirtualDeviceSpec()
+            cdrom_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.edit
+            cdrom_spec.device = cdrom_devices[0]
+
+        if cdrom_spec:
+            self.change_detected = True
+            self.configspec.deviceChange.append(cdrom_spec)
+
+    def configure_cdrom_list(self, vm_obj):
+        configured_cdroms = self.get_configured_cdroms()
+        cdrom_devices = self.get_vm_cdrom_devices(vm=vm_obj)
+        # configure IDE CD-ROMs
+        if configured_cdroms['ide'] is not None:
+            ide_device = None
+            ide_devices = self.get_vm_ide_devices(vm=vm_obj)
+            for expected_cdrom_spec in configured_cdroms['ide']:
+                ide_exist = False
+                if len(ide_devices) != 0:
+                    for device in ide_devices:
+                        if device.busNumber == expected_cdrom_spec['num']:
+                            ide_exist = True
+                            ide_device = device
+                            break
+                # if not find the matched ide controller or no existing ide controller
+                if not ide_exist or len(ide_devices) == 0:
+                    ide_ctl = self.device_helper.create_ide_controller(bus_number=expected_cdrom_spec['num'])
+                    ide_device = ide_ctl.device
+                    self.change_detected = True
+                    self.configspec.deviceChange.append(ide_ctl)
+
+                for cdrom in expected_cdrom_spec['cdrom']:
+                    if 'unit_number' not in cdrom or cdrom['unit_number'] not in [0, 1]:
+                        self.module.fail_json(msg="cdrom.unit_number is mandatory and valid value 0 or 1 for "
+                                                  "CD-ROM attached to IDE controller.")
+                    iso_path = cdrom['iso_path'] if 'iso_path' in cdrom else None
+                    unit_number = cdrom['unit_number']
+                    cdrom_exist = False
+                    if len(cdrom_devices) != 0:
+                        for target_cdrom in cdrom_devices:
+                            if target_cdrom.controllerKey == ide_device.key and target_cdrom.unitNumber == unit_number:
+                                cdrom_exist = True
+                                cdrom_device = target_cdrom
+                                break
+                    # create new CD-ROM
+                    if (not cdrom_exist or len(cdrom_devices) == 0) and ('state' not in cdrom or cdrom['state'] != 'absent'):
```

Comment:
I believe you should be able to replace `of (not cdrom_exist or len(cdrom_devices) == 0)...` with just `if not cdrom_device and cdrom.get('state') != 'absent'`

Reply:
Modified.